testcontainers-dapr/src/test/java/io/dapr/testcontainers/DaprContainerTest.java

```diff
@@ -0,0 +1,149 @@
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.testcontainers;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprClientBuilder;
+import io.dapr.client.domain.Metadata;
+import io.dapr.client.domain.State;
+import io.restassured.RestAssured;
+import io.restassured.path.json.JsonPath;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.testcontainers.Testcontainers;
+
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.configureFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static java.util.Collections.singletonMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class DaprContainerTest {
+
+  // Time-to-live for messages published.
+  private static final String MESSAGE_TTL_IN_SECONDS = "1000";
+  private static final String STATE_STORE_NAME = "kvstore";
+  private static final String KEY = "my-key";
+  private static final String PUBSUB_NAME = "pubsub";
+  private static final String PUBSUB_TOPIC_NAME = "topic";
+
+  @ClassRule
+  public static WireMockRule wireMockRule = new WireMockRule(wireMockConfig().port(8081));
+
+  @ClassRule
+  public static DaprContainer daprContainer = new DaprContainer("daprio/daprd")
+      .withAppName("dapr-app")
+      .withAppPort(8081)
+      .withAppChannelAddress("host.testcontainers.internal");
+
+  /**
+   * Sets the Dapr properties for the test.
+   */
+  @BeforeClass
+  public static void setDaprProperties() {
+    configStub();
+    Testcontainers.exposeHostPorts(8081);
+    System.setProperty("dapr.grpc.port", Integer.toString(daprContainer.getGrpcPort()));
```

Comment:
This is a global state and will affect other tests.

Reply:
yes.. Line 71 allows testcontainers started container to get back to the app
The System.setProperty lines set up the DaprClient to connect to the right port of the `daprd` sidecar started by testcontainers. 

Do you want us to move the configStub() call to each test? That will allow more control on newer tests, but I am not sure if you were suggesting something else here.

Reply:
I am just asking not to have global state that will pollute other tests. The solution can be anything that does keep the settings isolated in that context.